src/test/util_tests.cpp

```diff
@@ -1218,6 +1218,9 @@ BOOST_AUTO_TEST_CASE(test_LockDirectory)
     // has released the lock as we would expect by probing it.
     int processstatus;
     BOOST_CHECK_EQUAL(write(fd[1], &LockCommand, 1), 1);
+    // The following line invokes the ~CNetCleanup dtor without
+    // a paired SetupNetworking call. This is acceptable as long as
+    // ~CNetCleanup is a no-op for non-Windows platforms.
```

Comment:
Is this now only because of the change above? It's not obvious why calling `write()` would invoke a NetCleanup destructor, or why we end up in a state where we are trying to tear down networking, if it's never even been setup. In general it seems fragile to rely on ifdefs and code being a no-op to avoid additional issues.

Reply:
> Is this now only because of the change above?

This PR diff does not change behaviour of the commented LOC. However, it seems worth mentioning that currently the `SetupNetworking` and `~CNetCleanup` calls might be unbalanced. Perhaps, we might consider moving network initialization after daemonizing? But on platforms that support daemonizing, both calls are no-op...

> It's not obvious why calling `write()` would invoke a NetCleanup destructor...

Actually, it terminates the process with all consequences, including `~CNetCleanup` call.

Reply:
> However, it seems worth mentioning that currently the SetupNetworking and ~CNetCleanup calls might be unbalanced.

Maybe, but `SetupNetworking` and `CNetCleanup` only do anything on Windows, and the code here only runs for non-windows, where "unbalanced" calls to either doesn't matter, because there is no code being executed.

Reply:
> ... `SetupNetworking` and `CNetCleanup` only do anything on Windows, and the code here only runs for non-windows, where "unbalanced" calls to either doesn't matter, because there is no code being executed.

That is similar to what the comment says, right?

Reply:
I think my point is that we shouldn't need a comment like this, or it's not clear what value it provides. I would think it makes more sense to try and fix things so that we just don't have situations where we try and call network teardown code if we haven't setup any networking., rather than adding comments about Windows for code that only runs on Linux.